### PR TITLE
Publish dev version of package in PRs 

### PR DESCRIPTION
**Pull request by kongzii**
deploy please

### CODE REVIEW

File: .github/workflows/python_cd.yaml [hunk 1]
```diff
@@ -21,3 +23,21 @@ jobs:
       - name: Build and Publish
         run: poetry publish -p ${{ secrets.PYPI_TOKEN }} -u "__token__" --build
         shell: bash
+
+  publish-dev-package:
+    if: contains(github.event.pull_request.body, 'deploy') && github.event_name == 'pull_request'
```
Comment (kongzii) [inline .github/workflows/python_cd.yaml:28]: Initially, I wanted to run publish-dev-package after every commit, if mypy and pytest were successful. However even during my db_cache journey it already created so many versions that it's just spamming pypi.

I didn't find a better way than checking if there is a `deploy` word in the PR's description. If you know about something, I'm all ears! (no version of GPT helped more than this)

Screenshot by Dropbox Capture

Comment (kongzii) [reply .github/workflows/python_cd.yaml:28]: I thought there was some functionality to have jobs executed only after manually clicking on some "run" button in the pull request, but it seems like no! 🫨

Comment (gabrielfior) [reply .github/workflows/python_cd.yaml:28]: Curious if you have considered using `release please`? 
https://github.com/googleapis/release-please-action?tab=readme-ov-file#how-release-please-works

They seem to automate a lot of this stuff, and also have 1,7k stars on Github.

Comment (kongzii) [reply .github/workflows/python_cd.yaml:28]: @gabrielfior Did you use them before? I'm reading their doc for like 3th time but can't see where they automate this.

They also use GitHub action `on: push: branches: main`, but I don't see anything like `"release dev package if I add special  comment"` or anything similar.

Screenshot by Dropbox Capture

**Comment by kongzii** [comment]
deploy please

**Comment by kongzii** [comment]
deploy?

**Comment by kongzii** [comment]
Ah okay I'm not the only one frustrated about why this isn't working as I thought:

```
    if: github.event.issue.pull_request != null && github.event_name == 'issue_comment' && contains(github.event.comment.body, 'deploy please')
```

https://github.com/orgs/community/discussions/59389

So probably back to option 1), where it runs always after push to PR if the PR description contains 'deploy'

**Comment by kongzii** [comment]
deploy pleaseeeee

File: .github/workflows/python_cd.yaml [hunk 2]
```diff
@@ -3,9 +3,11 @@ name: Python CD
 on:
   release:
     types: [ published ]
+  pull_request_review_comment:
```
Comment (kongzii) [inline .github/workflows/python_cd.yaml:6]: Maybe if I comment on PR diff? deploy please this!

**Comment by kongzii** [comment]
Okay I give up! Backing up to the behavior from Friday -- if the PR description contains `deploy please`, then it will do the dev release on `pull_request: ` trigger (meaning pushes to the PR's branch)

**Review by gabrielfior — APPROVED** [review]

**Pull request merged**